examples/shader/Shader.cpp

```diff
@@ -343,28 +378,38 @@ class Geometry : public Effect
 ////////////////////////////////////////////////////////////
 int main()
 {
+    if (!sf::Shader::isAvailable())
+    {
+        std::cerr << "Shaders not supported on current system, aborting" << std::endl;
+        return EXIT_FAILURE;
+    }
+
     // Create the main window
     sf::RenderWindow window(sf::VideoMode({800, 600}), "SFML Shader", sf::Style::Titlebar | sf::Style::Close);
     window.setVerticalSyncEnabled(true);
 
-    // Load the application font and pass it to the Effect class
+    // Load the application font and store it as global variable for convenience
     const auto font = sf::Font::loadFromFile("resources/tuffy.ttf").value();
-    Effect::setFont(font);
 
     // Create the effects
-    Pixelate   pixelateEffect;
-    WaveBlur   waveBlurEffect;
-    StormBlink stormBlinkEffect;
-    Edge       edgeEffect;
-    Geometry   geometryEffect;
+    std::optional pixelateEffect   = Pixelate::tryLoad();
+    std::optional waveBlurEffect   = WaveBlur::tryLoad(font);
+    std::optional stormBlinkEffect = StormBlink::tryLoad();
+    std::optional edgeEffect       = Edge::tryLoad();
+    std::optional geometryEffect   = Geometry::tryLoad();
 
-    const std::array<Effect*, 5> effects{&pixelateEffect, &waveBlurEffect, &stormBlinkEffect, &edgeEffect, &geometryEffect};
+    const auto optionalToPtr = [&](auto& effect) -> Effect* { return effect.has_value() ? &*effect : nullptr; };
 
-    std::size_t current = 0;
+    std::array<Effect*, 5> effects{optionalToPtr(pixelateEffect),
+                                   optionalToPtr(waveBlurEffect),
+                                   optionalToPtr(stormBlinkEffect),
+                                   optionalToPtr(edgeEffect),
+                                   optionalToPtr(geometryEffect)};
 
-    // Initialize them
-    for (Effect* effect : effects)
-        effect->load();
+    const std::array<std::string, 5>
+        effectNames{"Pixelate", "Wave + Blur", "Storm + Blink", "Edge Post-effect", "Geometry Shader Billboards"};
+
+    std::size_t current = 0;
```

Comment:
Why did you get rid of the `getName` function? I think that's a better solution than a second array that owns the effect names.

Reply:
Because now I'm only conditionally creating the `Effect` objects if their loading succeeds, otherwise I'm storing `nullptr` for the ones that failed to load. The name has to be stored separately because of that.